src/query/execute_statement.c

```diff
@@ -11797,7 +11797,7 @@ do_insert_template (PARSER_CONTEXT * parser, DB_OTMPL ** otemplate, PT_NODE * st
   /* clear any previous error indicator because the rest of do_insert is sensitive to er_errid(). */
   er_clear ();
 
-  error = do_preliminary_checks (parser, statement, class_, update, value_clauses);
+  error = do_insert_checks (parser, statement, class_, update, &value_clauses);
```

Comment:
@lucianlazar2211 

It seems you wanted to modify `update` as `&update` instead of `value_clauses`?